grails-app/init/org/pih/warehouse/BootStrap.groovy

```diff
@@ -605,8 +605,10 @@ class BootStrap {
             liquibase = new Liquibase('views/drop-all-views.xml', new ClassLoaderResourceAccessor(), database)
             liquibase.update(null as Contexts, new LabelExpression());
 
-            // FIXME Not good that we'll need to update this on subsequent versions so this needs some more thought
-            List previousChangelogVersions = ["0.5.x", "0.6.x", "0.7.x", "0.8.x"]
+            // Find directories with names matching current versions pattern
+            List<String> changelogVersions = new File('grails-app/migrations').list().findAll { it.matches("[0-9]{1,3}.[0-9]{1,3}.x") }
```

Comment:
I just realized that you're looking at the changelog files rather than the database. We need to look at what changelogs have been executed already, not what could be executed.

Reply:
Sorry I confused this with the executedChangelogVersions above. So I think this is probably correct. I'm still worred about using the `File` API here because I don't know if that's going to work in all environments. 

I would either make use of `grailsResourceLocator`, `servletContext.getResourceAsStream` or something similar.

We use the former in LocalizationServer to pull in the appropriate message.properties for a locale.

Using the ServletContext would look something like this.

https://docs.grails.org/latest/ref/Servlet%20API/servletContext.html

Reply:
@awalkowiak I know we discussed this today but I couldn't tell if you meant that you wanted to merge this as-is or push it off to the future. I'm ok with either.